fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -270,19 +270,25 @@ public static void init() {
         // If we found a non-empty list of search parameter names to filter on,
         // then do the filtering. Otherwise, we're just going to return an empty list.
         if (includedSPs != null && !includedSPs.isEmpty()) {
-            // If "*" is contained in the included SP urls, then we can just return the unfiltered list
-            // now, since everything in the list will be included anyway.
-            if (includedSPs.contains(SearchConstants.WILDCARD_FILTER)) {
-                return unfilteredSearchParameters;
-            }
+            boolean includeAll = includedSPs.containsKey(SearchConstants.WILDCARD_FILTER);
 
-            // Otherwise, we'll walk through the unfiltered list and select the ones to be
-            // included in our result.
-            else {
-                for (SearchParameter sp : unfilteredSearchParameters) {
+            // Walk through the unfiltered list and select the ones to be included in our result.
+            for (SearchParameter sp : unfilteredSearchParameters) {
+                String code = sp.getCode().getValue();
+                String url = sp.getUrl().getValue();
 
-                    String url = sp.getUrl().getValue();
-                    if (includedSPs.contains(url)) {
+                if (includedSPs.containsKey(code)) {
+                    if (includedSPs.get(code).equals(url)) {
+                        results.add(sp);
+                    } else {
+                        log.info("Skipping search parameter with id='" + sp.getId() + "'. "
```

Comment:
had a hard time deciding whether this should be info-level or fine-level. opted for info-level since this behavior is new.
the issue with this is that we'll see these messages on each and every request that involves search parameters like this.

Reply:
Question: 
Executed on every search? if so, I'd say isLogLevel(fine)
If it's not, I'd say it's fine.

Reply:
I think it is executed for each search that targets a resource type with filtered parameters which match one of multiple possible target ids.  With our default config, we should never hit this.  The main motivator for keeping it info-level is that I think this will be the only log message you will get if you make an error while specifying the search parameter url in the config (so I don't want to hide that).
In the future, I think there is room for improving how we handle these search parameter filters.  For example, if we don't care about allowing dynamic updates to the config, we could check for all tenant configs during startup and validate the urls while we build a tenant-specific ParametersMap.

Reply:
Discussed with Paul and we think info level is ok for now given the above.